src/main/kotlin/test/ktortemplate/core/utils/pagination/TableExtensions.kt

```diff
@@ -1,62 +1,72 @@
 package test.ktortemplate.core.utils.pagination
 
-import org.jetbrains.exposed.sql.AutoIncColumnType
-import org.jetbrains.exposed.sql.BooleanColumnType
+import org.jetbrains.exposed.dao.id.EntityID
 import org.jetbrains.exposed.sql.Column
-import org.jetbrains.exposed.sql.EntityIDColumnType
 import org.jetbrains.exposed.sql.Expression
-import org.jetbrains.exposed.sql.IntegerColumnType
-import org.jetbrains.exposed.sql.LongColumnType
 import org.jetbrains.exposed.sql.Op
 import org.jetbrains.exposed.sql.Query
 import org.jetbrains.exposed.sql.SortOrder
 import org.jetbrains.exposed.sql.Table
-import org.jetbrains.exposed.sql.TextColumnType
-import org.jetbrains.exposed.sql.VarCharColumnType
 import org.jetbrains.exposed.sql.compoundAnd
 import org.jetbrains.exposed.sql.compoundOr
 import org.jetbrains.exposed.sql.select
 import org.jetbrains.exposed.sql.selectAll
+import java.util.UUID
 
 /**
- * Creates exposed query from list of FilterField
+ * Method that forces cast from Column<*> to Column<T>
  */
-fun Table.createFromFilters(filters: List<FilterField>): Query {
+fun <T> Column<*>.asType(): Column<T> {
+    @Suppress("UNCHECKED_CAST")
+    return this as Column<T>
+}
+
+/**
+ * Creates query from list of consumer filters.
+ */
+fun Table.fromFilters(filters: List<FilterField>): Query {
     val filtersOperations = this.createFilters(filters)
     return if (filtersOperations.isEmpty()) this.selectAll() else this.select { filtersOperations.compoundAnd() }
 }
 
 /**
- * Creates list of exposed operations from list of FilterField.
- * Add support to more fields when needed.
+ * Creates list of database operations from a list of consumer filters.
+ * Using columnType.valueFromDB() to get the underlying column data type.
  */
 fun Table.createFilters(filters: List<FilterField>): List<Op<Boolean>> {
     return filters.map { filterField ->
-        val column = this.columns.single { it.name == filterField.field }
-        when (column.columnType) {
-            is LongColumnType, is AutoIncColumnType, is EntityIDColumnType<*> -> {
-                column as Column<Long>
-                filterField.values.map { value -> Op.build { column.eq(value.toLong()) } }.compoundOr()
+        val column: Column<*> = this.columns.single { it.name == filterField.field }
+        val valueFromDB = column.columnType.valueFromDB(filterField.values.first()).let {
+            when (it) {
+                is EntityID<*> -> it.value
+                else -> it
+            }
+        }
+
+        // Support to more types can be added as needed bellow
+        when (valueFromDB) {
+            is Long -> column.asType<Long>().let {
```

Comment:
For each type we can safely convert it from **Column<*>** to **Column<Long>** using the extension method `asType()`